### PR TITLE
Simplify and fix master server ping logic

### LINKED ISSUE #15477: Lobbies seem joinable in playtest-20180729 for some minutes after the game has started
In playtest-20180729, though a game has started, the lobby still appears joinable in the server list. This persists for a couple mins or so.

### PR DESCRIPTION
**Pull request by pchote**
Fixes #15477 and removes a very old wart from the server code.

There are two key points to understanding this PR:
* `MasterServerPinger` historically tried to update the master server ASAP when something changes.  This is paired with a rate limit to stop the master server from being hammered: if the server data changes again while the first ping is in process, then the subsequent pings are ignored.  This is obviously a very bad idea, and historically only worked out of luck. A recent change (didn't follow up to find which one) changed the order that the `MasterServerPinger` is notified, and it now tries to send a ping immediately before the server state is updated.  The rate limiter then, more often than not, discards the ping containing the server state update (lobby &rarr; started or started &rarr; complete).  The stale state lingers for three minutes until the next heartbeat ping finally syncs the correct state, causing #15477.
* The server's run loop is driven by the `Socket.Select` call, which blocks until data is received from at least one connection or a given timeout (defined in microseconds) expires.  This means that there is no guarantee at all for when a `ServerTrait`'s `ITick` implementation is going to be called.  `TickTimeout` was added so that the `MasterServerPinger` could force the loop to run at least once every 3 minutes.  The authentication PR added a hack to reduce this to 100ms if it is waiting on a response to process.  On closer inspection, `MasterServerPinger`'s `TickTimeout` doesn't actually work, and the tick call runs every ~500ms anyway.

The first commit removes the unnecessary complexity of asking the ServerTraits for a timeout, and instead hardcodes a value of 1s normally or 0.1s when waiting for an auth callback.

The second commit moves the ping logic into `ITick.Tick` and change all the update triggers to set a timestamp.  We then use this timestamp to decide when to update the master server (and LAN clients), with duplicate pings in a single tick now sending the *final* state instead of something inconsistent in the middle.

### CODE REVIEW

**Comment by chrisforbes** [comment]
Gross. OK.

**Review by chrisforbes — APPROVED** [review]

**Comment by pchote** [comment]
It seems unlikely we'll get a second review on this in time, so taking as-is and I'll fix any potential regressions if they occur.

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/82b38f2d29be8d23a57390fe890cfa3be623509b

and picked to prep.

**Pull request merged**
